# DP: Make it so that a missing /etc/ld.so.conf does not cause an error
# DP: message, unless --verbose is enabled. This keeps the debian
# DP: installer from barfing during bootstrap of the system.
# DP:
# DP: Updated by gotom, the previous one seems being wrong because it supresses
# DP: _all_ warnings about "can't open configuration file". otoh, I introduce
# DP: newer patches.  it should be checked using chroot and should be contacted
# DP: to the upstream.
# DP: Date: (Updated 2005-01-02 gotom)

# previous patch.
#--- elf/ldconfig.c	2003-07-08 23:26:27.000000000 +0900
#+++ elf/ldconfig.c.debian	2003-07-08 23:29:43.000000000 +0900
#@@ -920,26 +920,24 @@
# {
#   FILE *file = NULL;
#   char *line = NULL;
#-  const char *canon;
#+  const char *canon = filename;
#   size_t len = 0;
#+  int file_fd;
# 
#   if (opt_chroot)
#     {
#       canon = chroot_canon (opt_chroot, filename);
#-      if (canon)
#-	file = fopen (canon, "r");
#-      else
#+      if (!canon)
# 	canon = filename;
#     }
#-  else
#-    {
#-      canon = filename;
#-      file = fopen (filename, "r");
#-    }
#+
#+  if ((file_fd = open(canon, O_RDONLY | O_EXCL, 0022)) != -1)
#+    file = fdopen (file_fd, "r");
# 
#   if (file == NULL)
#     {
#-      error (0, errno, _("Can't open configuration file %s"), canon);
#+      if (opt_verbose)
#+	error (0, errno, _("Can't open configuration file %s"), canon);
#       if (canon != filename)
# 	free ((char *) canon);
#       return;

2004-10-25  GOTO Masanori  <gotom@debian.or.jp>

        * elf/ldconfig.c: Don't print error when default ld.so.conf
          is not existed.

---
 elf/ldconfig.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/elf/ldconfig.c
+++ b/elf/ldconfig.c
@@ -1066,7 +1066,8 @@
 
   if (file == NULL)
     {
-      error (0, errno, _("Can't open configuration file %s"), canon);
+      if (strcmp(canon, LD_SO_CONF) != 0 || opt_verbose)
+	error (0, errno, _("Can't open configuration file %s"), canon);
       if (canon != filename)
 	free ((char *) canon);
       return;
